2_process/src/model_diagnostic.R

```diff
@@ -0,0 +1,40 @@
+# Load necessary packages
+library(dplyr)
+library(readr)
+library(stringr)
+library(sbtools)
+library(whisker)
+
+# Function to produce diagnostic log file
+get_diagnostic <- function(input_data, output_path) {
+  render_data <- list(pgdl_980mean = filter(input_data, model_type == 'pgdl', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
+                      dl_980mean = filter(input_data, model_type == 'dl', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_980mean = filter(input_data, model_type == 'pb', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
+                      dl_500mean = filter(input_data, model_type == 'dl', exper_id == "similar_500") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_500mean = filter(input_data, model_type == 'pb', exper_id == "similar_500") %>% pull(rmse) %>% mean %>% round(2),
+                      dl_100mean = filter(input_data, model_type == 'dl', exper_id == "similar_100") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_100mean = filter(input_data, model_type == 'pb', exper_id == "similar_100") %>% pull(rmse) %>% mean %>% round(2),
+                      pgdl_2mean = filter(input_data, model_type == 'pgdl', exper_id == "similar_2") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_2mean = filter(input_data, model_type == 'pb', exper_id == "similar_2") %>% pull(rmse) %>% mean %>% round(2))
+
+  template_1 <- 'resulted in mean RMSEs (means calculated as average of RMSEs from the five dataset iterations) of {{pgdl_980mean}}, {{dl_980mean}}, and {{pb_980mean}}°C for the PGDL, DL, and PB models, respectively.
+    The relative performance of DL vs PB depended on the amount of training data. The accuracy of Lake Mendota temperature predictions from the DL was better than PB when trained on 500 profiles
+    ({{dl_500mean}} and {{pb_500mean}}°C, respectively) or more, but worse than PB when training was reduced to 100 profiles ({{dl_100mean}} and {{pb_100mean}}°C respectively) or fewer.
+    The PGDL prediction accuracy was more robust compared to PB when only two profiles were provided for training ({{pgdl_2mean}} and {{pb_2mean}}°C, respectively). '
+
+  whisker.render(template_1 %>% str_remove_all('\n') %>% str_replace_all('  ', ' '), render_data ) %>% cat(file = file.path(output_path, 'model_diagnostic_text.txt'))
+}
+
+
+
+##### Get Model Diagnostics #####
+
+# Set workspace parameters 
+project_output_dir <- '2_process/out'
+
+# Load input data
+model_results <- readr::read_csv('2_process/out/model_summary_results.csv')
```

Comment:
This line is fine for now, but as a heads up, you won't be able to call `readr::read_csv` from your top-level "script" once we create that script and translate it into scipiper language. Go ahead and leave it for this PR but be open to changing your approach slightly in the next PR.

Reply:
Would I be able to call 'read.csv' instead? Is the readr:: command the issue?

Reply:
Actually, nevermind. I think I figured out a way around that. We'll see!

Reply:
The `::` is the big issue, yes...and actually, if you `include` the `readr` package you probably can call `read_csv` without the need for the `::`. You could definitely call `read.csv`.

One further consideration would be whether you _need_ to read the data in from file given that you already have it in memory once when you first create eval_data...but looking ahead to a shared-cache approach, as I noted in Teams, we'll end up often preferring a workflow that builds on files rather than R objects when the data need to be shared among people...so even though I had an R object target in mind as one option here, maybe the direction you're already headed is the most foresighted of all the possibilities.